### ISSUE TITLE
[basic.fundamental] Should we remove Note 1 on `int` having the "natural width"?

### ISSUE THREAD

**Issue by Eisenwave** [cplusplus/draft#6825]
> Plain `int`s are intended to have the natural width suggested by the architecture of the execution environment; the other signed integer types are provided to meet special needs.

[[basic.fundamental] Note 1](https://eel.is/c++draft/basic.fundamental#note-1)

I feel like this note is not doing much good at this point. Firstly, this sounds a lot like a *Recommended practice* paragraph, not like a note.

Besides that, it's not what implementers typically do, or should do. The "natural width" is the width of the general purpose register or the width of pointers; anything less will usually require zero- or sign-extension:
```cpp
char* advance(char* p, int x) {
    return p + x;
}
char* advance(char* p, long long x) {
    return p + x;
}
```
Clang emits:
```asm
advance(char*, int):
  movsxd rax, esi
  add rax, rdi
  ret
advance(char*, long long):
  lea rax, [rdi + rsi]
  ret
```

If implementers respected this note, `int` should be a 64-bit type on x86_64, which would make it "natural" and drop sign extensions. I don't believe this is useful or desirable; the general approach is to cap the `int` size at 32-bit, even if this makes it less natural and requires additional operations.

**Comment by eisenwave** [comment]
@jensmaurer thoughts on deleting the note? I think it's actually doing more harm than good.

In particular, the statement
> the other signed integer types are provided to meet special needs.

feels very wrong when `long` or `long long` have the natural width of a 64-bit architecture, while `int` meets the special need of having lower range and 32-bit compatibility, even if that requires sign/zero extension and adds cost.

In any case, I don't think we need a note that explains to implementers what `int` is, 40 years after C has been invented. This is just old gunk.

**Comment by jensmaurer** [comment]
Yes, let's drop this.  Please make a pull request and post a pointer to -core.
